pkg/cmd/experimental/syncgroups/schema1/memberextractor.go

```diff
@@ -0,0 +1,172 @@
+package schema1
+
+import (
+	"fmt"
+
+	"github.com/go-ldap/ldap"
+
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/config"
+)
+
+// New builds a new GroupMemberExtractor using a schema-appropriate config
+func New(config LDAPSyncConfig) GroupMemberExtractor {
```

Comment:
Don't use this config object.  I'm 99% sure you've build the wrong object.  Let me see the individual required objects flow into `New*` methods and we'll build back out from there.